.github/workflows/upload-ecr.yml

```diff
@@ -0,0 +1,86 @@
+name: Upload Image To ECR
+
+on:
+  push:
+    branches:
+      - '*deploy/*'
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+
+  upload-to-ecr:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Output variable names
+        id: vars
+        run: |
+          export BRANCH=$(echo "${GITHUB_REF#refs/*/}" | sed 's/.*\///')
+          echo ::set-output name=branch::$BRANCH
+          echo ::set-output name=short_sha::${GITHUB_SHA: -8}
+          export DEPLOYMENT=$(echo "${GITHUB_REF#refs/*/}" | sed 's/.*deploy\///')
+          echo ::set-output name=deployment::$DEPLOYMENT
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        env:
+          SESSION_NAME: ${{ steps.vars.outputs.branch }}-${{ steps.vars.outputs.short_sha }}
+        with:
+          aws-access-key-id: ${{ secrets.TEST_AWS_ACCESS_KEY_ID }}
```

Comment:
What is the TEST_ prefix about?

Reply:
this was isolated testing for the terraform-managed deploy IAM user + IAM role combo. since the existing ghactions newspeek IAM user also deploys on ECS it has an additional set of permissions granted, so I added a new set of credentials rather than overwriting those so that the other workflow would still work.